python/lsst/pipe/tasks/multiBand.py

```diff
@@ -818,14 +818,271 @@ def getSkySourceFootprints(self, mergedList, skyInfo, seed):
+        return parser
+
+    def __init__(self, butler=None, schema=None, peakSchema=None, **kwargs):
+        CmdLineTask.__init__(self, **kwargs)
+        if schema is None:
+            assert butler is not None, "Neither butler nor schema is defined"
+            schema = butler.get(self.config.coaddName + "Coadd_mergeDet_schema", immediate=True).schema
+        self.schemaMapper = afwTable.SchemaMapper(schema)
+        self.schemaMapper.addMinimalSchema(schema)
+        self.schema = self.schemaMapper.getOutputSchema()
+        if peakSchema is None:
+            assert butler is not None, "Neither butler nor peakSchema is defined"
+            peakSchema = butler.get(self.config.coaddName + "Coadd_peak_schema", immediate=True).schema
+
+        if self.config.simultaneous:
+            self.makeSubtask("multiBandDeblend", schema=self.schema, peakSchema=peakSchema)
+        else:
+            self.makeSubtask("singleBandDeblend", schema=self.schema, peakSchema=peakSchema)
+
+    def getSchemaCatalogs(self):
+        """Return a dict of empty catalogs for each catalog dataset produced by this task.
+
+        Returns
+        -------
+        result: dict
+            Dictionary of empty catalogs, with catalog names as keys.
+        """
+        catalog = afwTable.SourceCatalog(self.schema)
+        return {self.config.coaddName + "Coadd_deblendedFlux": catalog,
+                self.config.coaddName + "Coadd_deblendedModel": catalog}
+
+    def run(self, patchRefList, psfCache=100):
+        """Deblend the patch
+
+        Deblend each source simultaneously or separately
+        (depending on `DeblendCoaddSourcesTask.config.simultaneous`).
+        Set `is-primary` and related flags.
+        Propagate flags from individual visits.
+        Write the deblended sources out.
+
+        Parameters
+        ----------
+        patchRefList: list
+            List of data references for each filter
+        """
+        if self.config.simultaneous:
+            # Use SCARLET to simultaneously deblend across filters
+            msg = "Simultaneous deblending will be implemented once DM-15106 is completed"
+            raise NotImplementedError(msg)
+
+            # implement deblender here
+            # ...
+
+            # this is the proper code to write
+            fluxCatalogs, templateCatalogs = self.multibandBeblend.run(exposures, sources)
+            for n in len(patchRefList):
+                self.write(patchRefList[n], fluxCatalogs[n], templateCatalogs[n])
+        else:
+            # Use the singeband deblender to deblend each band separately
+            for patchRef in patchRefList:
+                exposure = patchRef.get(self.config.coaddName + "Coadd_calexp", immediate=True)
+                exposure.getPsf().setCacheCapacity(psfCache)
+                sources = self.readSources(patchRef)
+                self.singleBandDeblend.run(exposure, sources)
+                self.write(patchRef, sources)
```

Comment:
I think that this is a non-trivial change and someone who better understands `PipelineTask` is better equipped to decided how to split this up, if it can be done in a way that is consistent with other tasks.